Automattic/wp-calypso
# Fix linking with a social account from signup

Requires D6892

This PR fixes the redirect that wasn't working properly after you link your WordPress.com account with your google account on signup.

It also:
- removes the Facebook button which isn't used at the moment.
- updates the "Log In" link in the masterbar to use the new login page when enabled

### Testing Instructions
- Boot this branch locally or use calypso.live and navigate incognito to [/start/social](https://calypso.live/start/social?branch=fix/social-signup-2fa)
- Click on Connect with Google and select a google account which address matches the one you have for your WordPress.com account (have 2FA setup first for this account).
- Follow the social linking flow (need to enter your password and 2FA code)
- Ensure that you are redirected back to `/start/social` at the end (which should redirect you back to `/`)
- Test the resistance to errors by entering invalid inputs (email, password, 2fa codes...)
- Test that social linking, social login, logout still works (redirect) properly

### Reviews
